File: Segundo_Semestre/Programacao_Orientada_Objeto/aula_05_heranca/src/Funcionario.java
import java.util.Scanner;

public class Funcionario{
    private String nome;
    private int matricula;
    private float salario;

    public Funcionario(int matricula, String nome, float salario) {
        setMatricula(matricula);
        setNome(nome);
        setSalario(salario);
    }

    public Funcionario(){}

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getMatricula() {
        return matricula;
    }

    public void setMatricula(int matricula) {
        matricula = matricula;
    }

    public float getSalario() {
        return salario;
    }

    public void setSalario(float salario) {
        this.salario = salario;
    }

    public String toString(){
        String saida = "\nMatrícula: " + getMatricula();
        saida += "\nNome.....: " + getNome();
        saida += "\nSalário..: " + getSalario();
        return saida;
    }

    public void setStrig(){
        Scanner scan = new Scanner(System.in);
        System.out.print("\nDigite a matrícula: ");
        int matricula = scan.nextInt();
        scan.nextLine();
        System.out.print("Digite o nome: ");
        String nome = scan.nextLine();
        System.out.print("Digite o salário: ");
        float salario = scan.nextFloat();

        setMatricula(matricula);
        setNome(nome);
        setSalario(salario);
    }
}

File: Terceiro_Semestre/Desenvolvimento_Mobile/aula_05_conceitos_classe/exemplo_herancas/src/Funcionario.kt
open class Funcionario() {
    // Atributos com métodos setters
    var matricula: Int = 0
        set(value) {
            if (value > 0) field = value
        }
    var nome: String = ""
        set (value) {
            if (value != "") field = value
        }
    var salario: Double = 0.0
        set (value){
            if(value > 0.0) field = value
        }

    // Construtor
    constructor(mat: Int,nome: String, sal: Double):this(){
        matricula = mat
        this.nome = nome
        salario = sal
    }

    // Funções
    fun aumentoSalario ( porcentagem: Double) {
        if (porcentagem > 0.0){
            val valor = salario * porcentagem/100
        }
    }
    override fun toString() : String{
        return ("\nMatrícula: $matricula"
                + "\nNome.....: $nome"
                + "\nSalário..: $salario")
    }

}